docs/source/get_involved/code_of_conduct.rst

```diff
@@ -28,40 +29,59 @@ conform to the following Code of Conduct.
 1 Introduction
 --------------
 
-sktime is a community-oriented and -led project. We value the
-involvement of everyone in the community. We are committed to creating a
-friendly and respectful place for learning, teaching and contributing.
-All participants in our in-person events and online communications are
-expected to show respect and courtesy to others at all times.
+sktime is a community-oriented and -led project.
+
+We value the involvement of everyone in the community, and strive to be an open
+community. We commit to provide fair and revarding opportunities for everyone without
+restriction, to participate, learn, and become leaders of the community. We expect from
+members of our community to create such opportunities for others as have been created
+for them.
+
+We are committed to scientific quality and strive to be an open forum on matters
+of technical content and open community governance. All community members are expected
+to adhere to the principle of free speech in academia and science, in order to create
+the best possible science and the best possible open community.
```

Comment:
more a comment than any suggestion, but I think there is a fundamental tension between 'best possible science' and 'open community'.

- in my mind, 'best possible science' implies high standards which cannot reasonably be expected from the majority of contributors (or the majority of professional scientists in fact. top quality science is fundamentally hard!)
- there are personality traits (e.g. high anxiety or low self-esteem) which can make it difficult and very uncomfortable for people to receive critical feedback on a contribution - even if the criticism is constructive and would help them improve. This means there is trade-off between how high quality you want that individual contribution to be, versus, how welcome to you want that person to feel.

I do not know if there a solution to this. My own preference is to lean more towards being welcoming at the cost of 'best possible science'

Reply:
I think it's one of the trade-offs. We should be welcoming to early career contributors, and don't disadvantage them at all if something is not perfect. Conversely, we can reasonably expect of early career contributors to take on mentoring opportunities offered to them, and invest time in learning best practice, before wanting to radically change `sktime` core designs.
(e.g., everyone can be expected to read up on linting or how to use conda. We're happy to give pointers repeatedly and explain at lengths in the communtiy session, but we are not forced to merge PR that don't get over the code quality boundaries)

Reply:
Happy with explanation. Maybe worth rewording this sentence to make clear that we do not expect high absolute skills levels.

Reply:
yes, reworded it now that it's to the best of our abilities:

> to the best of their abilities, contribute to jointly creating
the best possible science and the best possible open community.